utils/preferences/src/preferences.rs

```diff
@@ -0,0 +1,30 @@
+use icu_locid::{
+    subtags::{Language, Region, Script},
+    Locale,
+};
+
+pub trait Preferences {
+    fn language(&self) -> &Language {
+        &Language::UND
+    }
+    fn script(&self) -> Option<&Script> {
+        None
+    }
+    fn region(&self) -> Option<&Region> {
+        None
+    }
```

Comment:
Nit: There is more than one concept of a region. There is the region in the sense of a localized version of a language, and there is a region in the sense of setting default preferences for things like currencies and measurement units. This distinction is handled automatically in BCP-47, but if you are forcing a locale into the Preferences mold, you should support both types of regions so that we don't lose information across the conversion.

Reply:
^ this question about language region versus preference region remains unresolved.

Reply:
I assume you refer to region subtag vs `rg` unicode extension.
TR35 calls the former `region` and the latter `region_override` with definition:

> A [Region Override](https://www.unicode.org/reports/tr35/#RegionOverride) specifies an alternate region to use for obtaining certain region-specific default values (those specified by the [<rgScope>](https://www.unicode.org/reports/tr35/tr35-info.html#rgScope) element), instead of using the region specified by the [unicode_region_subtag](https://www.unicode.org/reports/tr35/#unicode_region_subtag) in the Unicode Language Identifier (or inferred from the [unicode_language_subtag](https://www.unicode.org/reports/tr35/#unicode_language_subtag)).

I think we should follow.

Reply:
We need both types of region for data loading. For example, a British person living in the US who wants to make a MeasureFormat may want British translations ("metre") and American preferences ("miles per gallon") in the same constructor.

Reply:
I assumed that DP may want to have access to more than just LID. In that case we will pass `Preferences` (or `DataLocale::from(Preferences)` to `DP` rather than `Preferences::LID`. This will allow DP to reason about `region` and `region_override`.

Reply:
Where does the other type of region get stored? Does this basically mean that every preferences object has a region_override field?

Reply:
Every `Preference` that has `rg` as a relevant extension.

Reply:
Hmm. I'm not sure if I like codifying whether or not `rg` is a relevant extension in each and every `Preference` struct; I see that more as a data provider decision. CLDR could decide to change a certain piece of data from being language-based to region-based in a future edition; for example, currency symbols are currently being discussed as a possible candidate for this.

Reply:
> Hmm. I'm not sure if I like codifying whether or not rg is a relevant extension in each and every Preference struct; I see that more as a data provider decision.

I disagree. It is a potential Preferences field and each struct should decide if it is. I can see it being converted to `Option` in `DataLocale` where preferences that have relevant and present will use `Some` and everything else will result in `None`.

Reply:
In general, the set of what extension keys are relevant is dependent on the data keys reachable from a constructor. That set is changeable, so we should definitely at least make the preferences struct be non_exhaustive

Hmm, I wonder if there's a way to derive Preferences from the ResourceProvider bound?